### DISCUSSION TITLE
SRT not working listener mode.

### DISCUSSION THREAD

**Discussion by Rex-121** [HaishinKit/HaishinKit.swift#1399] [Q&A]
after `connection.open(URL(string: "srt://192.168.1.102:9889"), mode: .listener)
        stream.publish()`, there is no video send on OBS(which is set to "srt://192.168.1.102:9889?mode=caller"), but console says ` [com.haishinkit.SRTHaishinKit] [SRTSocket.swift:38] status > SRT Socket Connected`

**Comment by shogo4405** [comment]
At the time of creation, we confirmed that video packets were coming from OBS. However, there is no functionality to play back. Since the README.md is also confusing, I will remove it from the compatibility table.

**Comment by Rex-121** [comment]
Thank you for reply. Can I use Haishinkit stream to other device？ I mean，can i use stream to myself use your kit.

---- Replied Message ----
| From | ***@***.***> |
| Date | 04/01/2024 21:37 |
| To | ***@***.***> |
| Cc | Tyrant ***@***.***>,
Author ***@***.***> |
| Subject | Re: [shogo4405/HaishinKit.swift] SRT not working listener mode. (Discussion #1399) |

At the time of creation, we confirmed that video packets were coming from OBS. However, there is no functionality to play back. Since the README.md is also confusing, I will remove it from the compatibility table.

**Reply by Rex-121** [reply]
Because I still can't stream to OBS in Caller mode. But I notice that sound is streamed to OBS with no img

**Reply by shogo4405** [reply]
No way. HaishinKit can be used as a client to ingest into a server, similar to OBS. You need to set up a server like SRS.

**Comment by Rex-121** [comment]
Thanks again! I’m wondering do you have a plan to develop client to client or this can’t be done under SRT protocol?

---- Replied Message ----
| From | ***@***.***> |
| Date | 04/01/2024 22:47 |
| To | ***@***.***> |
| Cc | Tyrant ***@***.***>,
Author ***@***.***> |
| Subject | Re: [shogo4405/HaishinKit.swift] SRT not working listener mode. (Discussion #1399) |

No way. HaishinKit can be used as a client to ingest into a server, similar to OBS. You need to set up a server like SRS.

**Reply by shogo4405** [reply]
I have a basic understanding of the SRT protocol at the protocol level. Additionally, I understand that while there may be some issues at present, it is possible to receive packets as a listener in HaishinKit.

However, regarding the playback requested this time, the expected behavior would be for both audio and video to be displayed when the play method is called. This level of functionality, such as at the application layer, is not planned to be provided by HaishinKit.

**Reply by Rex-121** [reply]
Thanks for your time!

**Comment by shogo4405** [comment]
SRT Listener mode now avilable.
- https://github.com/shogo4405/HaishinKit.swift/pull/1558
